File: soapz-front/src/app/models/book.ts
import { BaseResponse } from "./baseResponse"

export interface BookResponse extends BaseResponse{
    book: BookInfo
}

export interface BookInfo{
    id: number,
    title: string,
    description: string
    yearPublished: number,
    isbn: string,
    pages: number,
    amount: number,
    publishingHouse: string,
    language: string,
    author: string,
    authorPhoto: string,
    authorDescription: string,
    photo: string,
    genres: string[]
}

File: soapz-back/SOAPZ-Reservation/Data/Book.cs
namespace SOAPZ_Reservation.Data
{
    public class Book
    {
        public int Id { get; set; }
        public string Title{ get; set; }
        public string Description { get; set; }
        public int YearPublished { get; set; }
        public string Isbn { get; set; }
        public int Pages { get; set; }
        public string? Photo { get; set; }
        public int Amount { get; set; }
        public int AuthorId { get; set; }
        public virtual Author Author { get; set; }
        public int PublishingHouseId { get; set; }
        public PublishingHouse PublishingHouse { get; set; }
        public int LanguageId { get; set; }
        public Language Language { get; set; }
        public ICollection<GenreList>? genre { get; set; }


    }
}
